src/main/java/net/clgd/ccemux/rendering/awt/AWTRenderer.java

```diff
@@ -225,51 +253,61 @@ private Point mapPointToCC(Point p) {
 		return new Point(x + 1, y + 1);
 	}
 
-	private boolean handleCtrlPress(char control) {
-		if (control == 't') {
-			computer.terminate();
-		} else if (control == 'r') {
-			if (!computer.isOn()) {
-				computer.turnOn();
-			} else {
-				computer.reboot();
-			}
-		} else if (control == 's') {
-			computer.shutdown();
-		} else if (control == 'v') {
-			try {
-				computer.paste(
-						(String) Toolkit.getDefaultToolkit().getSystemClipboard().getData(DataFlavor.stringFlavor));
-			} catch (HeadlessException | UnsupportedFlavorException | IOException e) {
-				log.error("Could not read clipboard", e);
-			}
-		} else {
-			return false;
-		}
-
-		return true;
+	/**
+	 * Determine whether {@code key} and {@code char} events should be queued.
+	 *
+	 * If any of the action keys are pressed (terminate, shutdown, reboot) then such events will
+	 * be blocked.
+	 *
+	 * @return Whether such events should be queued.
+	 */
+	private boolean allowKeyEvents() {
+		return shutdownTimer < 0 && rebootTimer < 0 && terminateTimer < 0;
 	}
 
 	@Override
 	public void keyTyped(KeyEvent e) {
-		if (isPrintableChar(e.getKeyChar())) {
+		if (isPrintableChar(e.getKeyChar()) & allowKeyEvents()) {
 			computer.pressChar(e.getKeyChar());
 			blinkLockedTime = 0.25d;
 		}
 	}
 
 	@Override
 	public void keyPressed(KeyEvent e) {
-		computer.pressKey(translateToCC(e.getKeyCode()), false);
+		if ((e.getModifiers() & Toolkit.getDefaultToolkit().getMenuShortcutKeyMask()) != 0) {
+			char real = (char) (e.getKeyChar() + 96);
+
+			// Start action timers
+			if (real == 's' && shutdownTimer < 0) shutdownTimer = 0;
+			if (real == 'r' && rebootTimer < 0) rebootTimer = 0;
+			if (real == 't' && terminateTimer < 0) terminateTimer = 0;
+
+			// Handle pasting, this exists early.
+			if (real == 'v') {
```

Comment:
The only reason why we used a bunch of `if` statements here previously, was because Xtend didn't allow us to use a `switch` for reasons I don't remember. Though, now that we need to check the timers as well, I don't know if a `switch` is still worth it.